service/history/api/updateworkflowoptions/api.go

```diff
@@ -138,10 +151,30 @@ func mergeWorkflowExecutionOptions(
 		return nil, err
 	}
 	updateFields := util.ParseFieldMask(updateMask)
-	if _, ok := updateFields["versioningOverride"]; ok {
-		mergeInto.VersioningOverride = mergeFrom.GetVersioningOverride()
+
+	// ==== Priority
+
+	if _, ok := updateFields["priority"]; ok {
+		mergeInto.Priority = mergeFrom.GetPriority()
+	}
+
+	if _, ok := updateFields["priority.priorityKey"]; ok {
+		mergeInto.Priority = cmp.Or(mergeInto.Priority, &commonpb.Priority{})
```

Comment:
hmm, maybe this should do an `if` like the updateactivityoptions one to avoid the extra allocation? (I'm not sure it does an extra allocation but it looks like it)

if you want a generic helper to initialize a pointer if it's nil, I'd be okay with that. I've wanted it in various places but it feels a little un-go-like so I've just done if statements or local helpers